workflow/validate/validate_test.go

```diff
@@ -1670,19 +1671,49 @@ spec:
     resource:
       action: foo
       manifest: |
-      apiVersion: v1
-      kind: ConfigMap
-      metadata:
-        name: whalesay-cm
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: whalesay-cm
 `
 
 // TestInvalidResourceWorkflow verifies an error against a workflow of an invalid resource.
 func TestInvalidResourceWorkflow(t *testing.T) {
 	wf := unmarshalWf(invalidResourceWorkflow)
 	err := ValidateWorkflow(wfClientset, metav1.NamespaceDefault, wf, ValidateOpts{})
-	assert.Error(t, err, "templates.whalesay.resource.manifest must be a valid yaml")
+	assert.Equal(t, "templates.whalesay.resource.manifest must be a valid yaml", err.Error())
 
 	wf = unmarshalWf(invalidActionResourceWorkflow)
 	err = ValidateWorkflow(wfClientset, metav1.NamespaceDefault, wf, ValidateOpts{})
-	assert.Error(t, err, "templates.whalesay.resource.action must be either get, create, apply, delete or replace")
+	assert.Equal(t, "templates.whalesay.resource.action must be one of: get, create, apply, delete, replace, patch", err.Error())
+}
+
+var invalidPodGC = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: pod-gc-strategy-unknown-
+spec:
+  podGC:
+    strategy: Foo
+  entrypoint: whalesay
+  templates:
+  - name: whalesay
+    container:
+      image: docker/whalesay:latest
+      command: [cowsay]
+      args: ["hello world"]
+`
+
+// TestUnknownPodGCStrategy verifies pod gc strategy is correct.
+func TestUnknownPodGCStrategy(t *testing.T) {
+	wf := unmarshalWf(invalidPodGC)
+	err := ValidateWorkflow(wfClientset, metav1.NamespaceDefault, wf, ValidateOpts{})
+	assert.Equal(t, "podGC.strategy unknown strategy 'Foo'", err.Error())
```

Comment:
Is there any reason why you didn't use `EqualError`?

Reply:
Didn't know EqualError existed. Thanks for that. I fixed it.